binary_search_tree/tree.py

```diff
@@ -11,38 +11,111 @@ def __init__(self, key, val = None):
 
 
 class Tree:
-    def __init__(self):
-        self.root = None
+    def __init__(self, root=None):
+        self.root = root
+    def __init__(self, root=None):
+        self.root = root
+    def add_helper(self, current, key, value):
+        if current == None:
+            return TreeNode(key, value)
+        elif current.key >= key:
+            current.left = self.add_helper(current.left, key, value)
+        else:
+            current.right = self.add_helper(current.right, key, value)
+        return current
+    # Time Complexity: 0(log n)
+    # Space Complexity: 0(log n)
+    def add(self, key, value =None):
+        if self.root == None:
+            self.root = TreeNode(key, value)
+        else:
+            self.add_helper(self.root, key, value)
 
-    # Time Complexity: 
-    # Space Complexity: 
-    def add(self, key, value = None):
-        pass
-
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: 0(log n)
+    # Space Complexity: 0(log n)
+    def find_helper(self, current, key):
+        if current.key == key:
+            return current.value
+        if current.key >= key:
+            if not current.left:
+                return None
+            else:
+                return self.find_helper(current.left, key)
+        else:
+            if not current.right:
+                return None
+            else:
+                return self.find_helper(current.right, key)
     def find(self, key):
-        pass
+        if self.root == None:
+            return None
+        else:
+            return self.find_helper(self.root, key)
+
+    def inorder_helper(self, current, items):
+        if current is not None:
+            self.inorder_helper(current.left, items)
+            items.append({"key": current.key, "value": current.value})
+            self.inorder_helper(current.right, items)
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: 0(n)
+    # Space Complexity: 0(n)
+    def inorder_helper(self, current, items):
+        if current is not None:
+            self.inorder_helper(current.left, items)
+            items.append({"key": current.key, "value": current.value})
+            self.inorder_helper(current.right, items)
+    
     def inorder(self):
-        pass
+        items = []
```

Comment:
Make sure your naming convention stays consistent. Looks like you named this variable `tree` below